crypto/src/main/kotlin/net/corda/v5/crypto/CryptoUtils.kt

```diff
@@ -97,7 +97,7 @@ fun PublicKey.isFulfilledBy(otherKeys: Iterable<PublicKey>): Boolean =
  * [otherKeys] is a [CompositeKey], this function will not find a match.</i>
  */
 fun PublicKey.containsAny(otherKeys: Iterable<PublicKey>): Boolean {
-    return if (this is CompositeKey) keys.intersect(otherKeys).isNotEmpty()
+    return if (this is CompositeKey) otherKeys.any { keys.contains(it) }
```

Comment:
Would it make sense to add a `val keys = this.keys` above this so we don't rerun it on every lambda call?

Reply:
Or disregard please, not overhead in still doing it anyway.

Reply:
The keys extension for PublicKey (or the `leafkeys` val) should probably be lazy so we don't flatmap the children collection in a tight loop? If that was a hash set fixed for the runtime of this method, this implementation would be good. If `this.keys` has non-trivial runtime, less so.

Reply:
I'm not entirely sure what your suggestion is. Could you elaborate further?

Reply:
I think this is a good observation. `leafKeys` is internal so let's not worry about this here?

We should be getting full set of keys though to look into.

But maybe after all let's do a `val keys = this.keys` after all? Because `getKeys()` might delegate to `CompositeKey.getLeafKeys()` which might be doing whatever under the hood like recomputing things.

Reply:
I ended up just exporting the `this.keys` to a value. Making `leafKeys` lazy is something we could discuss later on, not sure if it is possible though. What happens if the underlying keys change? Is that a regular thing in a composite key? In that case `lazy` is not an option because it will only be evaluated once.